### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by manuel12**
Fixes: https://github.com/gbowne1/reactsocialnetwork/issues/212

This PR includes:
- The express backend(located in server folder).
- The react  front-end(now located in client folder), with updated components to fetch and save data on the backend, instead of using localStorage.
- Updated e2e tests.
- Updated readme to reflect folder related to running tests.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "test:cypress": "cd ../e2e/ && npx cypress run",
+    "test:cypress:open": "cd ../e2e/ && npx cypress open",
+    "eject": "react-scripts eject",
+    "format": "npx prettier --write ."
+  },
+  "eslintConfig": {
```
Comment (pawel975) [inline client/package.json:45]: This eslint config section should be omitted because it throws errors. I deleted it in last eslint fix PR.

**Review by pawel975 — APPROVED** [review]
Approve changes 😀

**Review by gbowne1 — APPROVED** [review]
This PR has merge conflicts, but I have reviewed the code visually  Approving for merge.

**Comment by gbowne1** [comment]
This PR has merge conflicts @manuel12 but otherwise I am approving for merge.

File: client/package.json (hunk 1 above)
Comment (manuel12) [reply client/package.json:45]: Thanks, will remove it :)

**Comment by manuel12** [comment]
> This PR has merge conflicts @manuel12 but otherwise I am approving for merge.

Will solve conflicts when I have sometime, but probably after checking some of the previous open PRs.

**Pull request merged**